helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -97,7 +111,17 @@ public CustomRestClientImpl(HttpClient httpClient) {
           kv -> result.put(kv.getKey(), kv.getValue().get(IS_HEALTHY_FIELD).asBoolean()));
       return result;
     };
-    return handleResponse(post(url, payLoads), jsonConverter);
+    try {
+      // use semaphore to allow X threads to submit HttpRequests at a time
+      // without thread synchronization, duplicate http requests will still exist
```

Comment:
> duplicate http requests will still exist

So is there another way of preventing this completely? 
I might be lack of context here, but the current method limits parallelism for dedup, right? Meaning we will limit the threads even system has enough resource. This is not a scalable solution.
Can we use a concurrent map to hold all the in-progress query with some key and avoid the queries with the same keys to dedup?